test/org/zalando/stups/kio/unit_test/api_test.clj

```diff
@@ -95,23 +95,55 @@
 
 (deftest ^:unit test-require-write-access
   (facts "write access"
-    (fact "a robot does not get access if the uid is missing"
+    (fact "access is denied if the uid is missing"
       (api/require-write-authorization .request. .team.) => (throws Exception anything (with-status? 403))
       (provided
         .request. =contains=> {:tokeninfo     {"realm" "/services"
-                                               "scope" ["uid" "application.write_all"]}
-                               :configuration {:allowed-uids "stups_robot,foo"}}))
+                                               "scope" ["uid" "application.write"]}
+                               :configuration {:admin-users "/services/stups_robot,foo"}}))
 
-    (fact "a robot does get access if it has write_all scope and required uid"
+    (fact "access is denied if the realm is neither services nor employees"
+      (api/require-write-authorization .request. .team.) => (throws Exception anything (with-status? 403))
+      (provided
+        .request. =contains=> {:tokeninfo     {"realm" "/somerealm"
+                                               "scope" ["uid" "application.write"]}
+                               :configuration {:admin-users "/services/stups_robot,foo"}}))
+
+    (fact "a robot does get access if it has write scope and required uid"
       (api/require-write-authorization .request. .team.) => nil
       (provided
         .request. =contains=> {:tokeninfo     {"realm" "/services"
                                                "uid"   "stups_robot"
-                                               "scope" ["uid" "application.write_all"]}
-                               :configuration {:allowed-uids "stups_robot,foo"}}
+                                               "scope" ["uid" "application.write"]}
+                               :configuration {:admin-users "stups_robot,foo"}}
```

Comment:
I don't get this test case. If a robot has write scope, it doesn't need a special uid anyways? And if it has a special uid, what is the write scope for? Also, config misses the realm but is probably not necessary anyways